src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -507,7 +507,7 @@ private static string GenerateUpgradeReport(NuGetProject nuGetProject, string ba
             TelemetryEvent ToTelemetryPackage(Tuple<string, string> package)
             {
                 var subEvent = new TelemetryEvent(eventName: null);
-                subEvent.AddPiiData("id", package.Item1);
+                subEvent.AddPiiData("id", package.Item1?.ToLowerInvariant() ?? "(empty package id)");
```

Comment:
When can the id be empty/

Reply:
I don't know of a specific scenario when it can be empty.  However, it is attacker controlled data and with this change I am going to be dereferencing it.  The code is inside of a finally block, so the process should not fall down, but a null-ref exception here would mean telemetry would be lost and this would be difficult to detect.

Reply:
I'd be willing to guess that it's impossible for that value to be null. NuGet would've crapped out way earlier :)